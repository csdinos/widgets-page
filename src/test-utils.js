import React from 'react'
import {render as rtlRender} from '@testing-library/react'
import {createStore} from 'redux'
import {Provider} from 'react-redux'
import {reducer} from './reducers'

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>
  }

  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

export const initialState = {
  widgets: [],
  newWidget: {
    id: -1,
    name: '',
    language: '',
  },
  deleteModalId: null,
  hasLoaded: false,
  shouldUpdateStorage: false
}

export * from '@testing-library/react'
export {render}