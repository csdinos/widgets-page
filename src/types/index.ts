export type Widget = {
  id: number;
  name: string;
  language: string;
}
export type State = {
  widgets: Array<Widget>
  newWidget: Widget
  deleteModalId: number | null,
  hasLoaded: boolean,
  shouldUpdateStorage: boolean
}