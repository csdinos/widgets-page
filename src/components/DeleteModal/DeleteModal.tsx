import {Button, Modal, Typography} from '@material-ui/core'
import {makeStyles, Theme, createStyles} from '@material-ui/core/styles'
import {State} from '../../types'
import {useDispatch, useSelector} from 'react-redux'
import * as actions from '../../actions'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    modal: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
    },
    paper: {
      textAlign: 'center',
      width: 400,
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
  }),
)

export const DeleteModal = () => {
  const classes = useStyles()
  const dispatch = useDispatch()
  const id = useSelector((state: State) => state.deleteModalId)

  const onClose = () => {
    dispatch(actions.setIdForDeleteModal(null))
  }

  const onConfirmation = () => {
    dispatch(actions.removeWidget(id))
    dispatch(actions.setIdForDeleteModal(null))
    dispatch(actions.setShouldUpdateStorage(true))
  }

  return (
    <Modal
      open={!!id}
      onClose={onClose}
      className={classes.modal}
    >
      <div className={classes.paper}>
        <Typography variant="body1" gutterBottom>Are you sure?</Typography>
        <Button
          id="confirm-delete-btn"
          variant="contained"
          color="secondary"
          onClick={onConfirmation}
        >Delete</Button>
      </div>
    </Modal>
  )
}