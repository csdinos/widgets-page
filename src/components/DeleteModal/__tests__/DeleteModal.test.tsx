import React from "react";
import {initialState, render} from '../../../test-utils'
import {DeleteModal} from "../DeleteModal";
import {State} from '../../../types'

const state: State = {
  ...initialState,
  widgets: [{
    name: 'Doge',
    language: 'Meme',
    id: 1994
  }],
  hasLoaded: true,
  deleteModalId: 1994
}

describe("<DeleteModal />", () => {
  test("should display the DeleteModal", async () => {
    const { asFragment } = render(<DeleteModal />, {initialState: state});
    expect(asFragment()).toMatchSnapshot()
  });
});