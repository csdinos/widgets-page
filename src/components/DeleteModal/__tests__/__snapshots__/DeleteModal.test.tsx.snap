// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteModal /> should display the DeleteModal 1`] = `<DocumentFragment />`;
